Framework/PythonInterface/plugins/algorithms/SetSampleFromLogs.py

```diff
@@ -0,0 +1,167 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.api import (AlgorithmFactory, DistributedDataProcessorAlgorithm)
+from mantid.kernel import (ConfigService)
+from mantid import _kernel as mtd_kernel
+from mantid.simpleapi import (SetSample)
+
+
+PROPS_FOR_SETSAMPLE = ["InputWorkspace", "Geometry", "Material", "Environment", "ContainerGeometry", "ContainerMaterial"]
+
+
+def _toDict(propertyManager):
+    result = {}
+    if propertyManager:
+        for key in propertyManager.keys():
+            result[key] = propertyManager[key].value
+            if isinstance(result[key], mtd_kernel.std_vector_dbl):
+                result[key] = list(result[key])
+    return result
+
+
+def _findKey(dictlike, *args):
+    for name in args:
+        if (name in dictlike) and bool(dictlike[name]):
+            return name
+    return ''  # indicates failure
+
+
+def _hasValue(dictLike, key):  # TODO refactor into *args and return the one that exists or empty string
+    return (key in dictLike) and bool(dictLike[key])
+
+
+def _getLogValue(propertyManager, key):
+    if not _hasValue(propertyManager, key):
+        raise ValueError('Failed to find "{}" in logs'.format(key))
```

Comment:
I suppose fail early is a good practice to follow.
However, some user might find it annoying if their entire workflow blows up because one data file with a corrupted log entry.

Reply:
Since algorithms are atomic, it is hard to raise an exception before they are run. This particular message should be fairly clear as to what happened, but will still be deep into a workflow when the failure happens due to a missing log.